src/zmqpp/loop.hpp

```diff
@@ -59,8 +59,10 @@ namespace zmqpp
          * \param socket the socket to monitor.
          * \param callable the function that will be called by the loop when a registered event occurs on socket.
          * \param event the event flags to monitor on the socket.
+         * \param after_remove_cb will be called by loop after remove() completion.
+         *        See tests/test_loop.cpp: socket_closed_in_timer and socket_closed_after_remove_at_timer
          */
-        void add(socket_t& socket, Callable callable, short const event = poller::poll_in);
+        void add(socket_t& socket, Callable callable, short const event = poller::poll_in, Callable after_remove_cb = Callable(nullptr));
```

Comment:
will this break ABI?

Reply:
Yes. Method's name will be mangled differently. Clients of libzmqpp need to be recompiled. But no changes at source code of clients.